### PR TITLE
feat(lightning): Implemented The "Quick Setup" Lightning Channel Open Flow

### PR DESCRIPTION
**Pull request by coreyphillips**
This PR implements the logic for the "Quick Setup" lightning channel open flow with Blocktank

To test, ensure you have a sufficient amount of on-chain regtest coins. Once on-chain regtest coins are available to the app, tap the "Pay Instantly" suggestion card or the "Transfer" button in the Wallet Details view and proceed with the "Quick Setup" option.

At this time, the regtest server only allows for inbound liquidity. Any attempt to create a channel with outbound liquidity by using the slider and setting "Spending Balance" to be anything, but zero, will result in an error. Once an order is paid, the app will watch for the order and finalize it in the background once Blocktank sees the transaction confirm. Shortly after it's finalized, Blocktank will open the channel and it will become present in the lightning channel list.

https://user-images.githubusercontent.com/8532651/189009935-21473a5c-9653-4951-af19-9e3515cc8d97.mp4

### CODE REVIEW

**Comment by MiguelMedeiros** [comment]
Awesome @coreyphillips ! :+1: 
Is it possible to add a `Lightning Connection` event to the `Activity log`?

## Preview 

image

**Comment by coreyphillips** [comment]
> Awesome @coreyphillips ! 👍 Is it possible to add a `Lightning Connection` event to the `Activity log`?
> 
> ## Preview
> image

That will be a separate PR. Currently lightning transactions will not populate the `Activity Log`.

**Review by limpbrains — APPROVED** [review]

File: src/screens/Wallets/WalletsDetail/BitcoinBreakdown.tsx [hunk 1]
```diff
@@ -63,7 +66,10 @@ const BitcoinBreakdown = (): ReactElement => {
 			/>
 			<View color={'transparent'} style={styles.transferRow}>
 				<View color={'gray4'} style={styles.line} />
-				<TouchableOpacity>
+				<TouchableOpacity
+					onPress={(): void => {
+						navigation.navigate('Introduction');
```
Comment (pwltr) [inline src/screens/Wallets/WalletsDetail/BitcoinBreakdown.tsx]: After https://github.com/synonymdev/bitkit/pull/340 is merged we can also navigate to the nested navigator. Would save us from having the Lightning flow screens in `RootNavigator` as well.

```suggestion
navigation.navigate('LightningRoot', {
	screen: 'Introduction',
});
```

Comment (coreyphillips) [reply src/screens/Wallets/WalletsDetail/BitcoinBreakdown.tsx]: Updated [here](https://github.com/synonymdev/bitkit/pull/349/commits/31594bb40007ca4eb5f82b9298df0eae2d4194f2).

File: src/navigation/root/RootNavigator.tsx [hunk 2]
```diff
@@ -35,6 +35,11 @@ import Contacts from '../../screens/Contacts/Contacts';
 import Contact from '../../screens/Contacts/Contact';
 import ContactEdit from '../../screens/Contacts/ContactEdit';
 import type { RootStackParamList } from '../types';
+import Introduction from '../../screens/Lightning/Introduction';
+import QuickSetup from '../../screens/Lightning/QuickSetup';
+import CustomSetup from '../../screens/Lightning/CustomSetup';
+import QuickConfirm from '../../screens/Lightning/QuickConfirm';
+import Result from '../../screens/Lightning/Result';
```
Comment (pwltr) [inline src/navigation/root/RootNavigator.tsx]: Can remove these now

**Review by pwltr — CHANGES_REQUESTED** [review]

File: src/navigation/root/RootNavigator.tsx (hunk 2 above)
Comment (coreyphillips) [reply src/navigation/root/RootNavigator.tsx]: Ah, thanks. Removed [here](https://github.com/synonymdev/bitkit/pull/349/commits/18bf934548a484d5b67c706044128830607a13be).
I also temporarily added `LightningRoot` to the `SettingsNavigator` for the `Channels` component until we implement opening channels to nodes other than Blocktank.

**Review by pwltr — APPROVED** [review]

**Pull request merged**